### PR TITLE
feat(metrics): adding option to automatically collect metrics from k8s cluster

### PR DESCRIPTION
**Pull request by tvaintrob**

### CODE REVIEW

File: charts/cluster-agent/Chart.yaml [hunk 1]
```diff
@@ -13,3 +13,30 @@ version: 0.1.10
...
+    version: 13.8.0
+
+  - name: prometheus-mongodb-exporter
+    alias: mongodb-metrics
+    repository: https://prometheus-community.github.io/helm-charts/
+    condition: metrics.mongodb.enabled
+    version: 2.8.1
+
+  - name: redis-metrics
+    alias: epsagon-redis-exporter
```
Comment (haimrait) [inline charts/cluster-agent/Chart.yaml:33]: Change name -> alias

File: charts/cluster-agent/README.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 # Epsagon Agent
 
-![Version: 0.1.5](https://img.shields.io/badge/Version-0.1.5-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.0.2](https://img.shields.io/badge/AppVersion-1.0.2-informational?style=flat-square)
+![Version: 0.1.10](https://img.shields.io/badge/Version-0.1.10-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.0.5](https://img.shields.io/badge/AppVersion-1.0.5-informational?style=flat-square)
```
Comment (haimrait) [inline charts/cluster-agent/README.md:10]: Why did you change the version?

File: charts/cluster-agent/README.md [hunk 3]
```diff
@@ -21,6 +21,15 @@ helm repo update
...
 Kubernetes 1.16+
 
+## Requirements
+
+| Repository | Name | Version |
+|------------|------|---------|
+| https://prometheus-community.github.io/helm-charts/ | metrics-agent(prometheus) | 13.8.0 |
+| https://prometheus-community.github.io/helm-charts/ | mongodb-metrics(prometheus-mongodb-exporter) | 2.8.1 |
+| https://prometheus-community.github.io/helm-charts/ | rabbitmq-metrics(prometheus-rabbitmq-exporter) | 1.0.0 |
+| https://prometheus-community.github.io/helm-charts/ | epsagon-redis-exporter(redis-metrics) | 4.0.2 |
```
Comment (haimrait) [inline charts/cluster-agent/README.md:31]: epsagon-redis-exporter(redis-metrics) | 4.0.2 | i think you have a missmatch between exporter and the name

File: charts/cluster-agent/values.yaml [hunk 4]
```diff
@@ -69,3 +71,59 @@ clusterAgent:
...
+  serviceAccounts:
+    alertmanager:
+      create: false
+    nodeExporter:
+      create: false
+    pushgateway:
+      create: false
+
+  server:
+    remoteWrite: []
```
Comment (Alon-Katz) [inline charts/cluster-agent/values.yaml:111]: How does the remote write with the user token and cluster name works?

File: charts/cluster-agent/README.md (hunk 2 above)
Comment (tvaintrob) [reply charts/cluster-agent/README.md:10]: I didn't, the readme is auto generated

File: charts/cluster-agent/values.yaml (hunk 4 above)
Comment (tvaintrob) [reply charts/cluster-agent/values.yaml:111]: metrics-agent:
  server:
    remoteWrite:
      - url: https://dev.collector.epsagon.com/ingestion?d6796942-bac6-47bd-b985-db811c137d86
        basic_auth:
          username: "d6796942-bac6-47bd-b985-db811c137d86"

Comment (Alon-Katz) [reply charts/cluster-agent/values.yaml:111]: but it supposed to be configured automatically for the user and not manually, and the cluster name should also be added to the remote write from the set params the user gave when he installed the chart.

Comment (tvaintrob) [reply charts/cluster-agent/values.yaml:111]: cant be configured automatically, because we cant use a value in the values file

Comment (tvaintrob) [reply charts/cluster-agent/values.yaml:111]: open to ideas

Comment (Alon-Katz) [reply charts/cluster-agent/values.yaml:111]: Worst case is forking the prometheus helm chart and changing where the remote write value is used to a custom variable that we will define in the .tpl file or checking if the tpl file can override existing values. I just think that without this option we lose the whole point of making a single helm chart, if the user still needs to create a local values file and fill all the information or using a huge set commands

Comment (tvaintrob) [reply charts/cluster-agent/values.yaml:111]: i agree, ill look into forking the prometheus chart, but i think its out of the scope of this pr, merging this for now, and in the near future we'll fork the prom chart to avoid the remote write
